paper.md

```diff
@@ -0,0 +1,130 @@
+languages as well). For generality we refer to the software package being
+built as the "target". The CMake ecosystem contains a number of tools which
+facilitate the development of the target, but notably absent from the CMake
+ecosystem are resources to facilitate the development of the target's
+literal build system. From a historical perspective, this is understandable
+since CMake-based build systems have tended to be relatively small (*i.e.*,
+less than ~1K lines of code) and tightly coupled to the identity of the
+target. Modern build systems are increasingly complex and, depending on the
+needs of the target, can include: hardware/software introspection, managing
+optional dependencies, managing the target's development environment,
+supporting multiple coding languages, coding language introspection, etc.
+While it may be argued that some of these tasks are better handled outside
+the build system, the point remains that CMake is actually a very flexible
+language which can be used to automate complex development tasks. In all
+likelihood scientific software developers are going to continue to use
+CMake for increasingly complex tasks and it behooves the research community
+to grow the CMake ecosystem accordingly.
+
+One of the most basic elements of a software development ecosystem is the
+ability to generate API documentation.
+Anecdotal evidence [@official_solution] indicates that the developers of the
+CMake language (*i.e.*, Kitware) internally write their API documentation
+using reST and Sphinx. Following best practices, this reST API 
+documentation resides next to the CMake code being described. Kitware has
+also written a Sphinx plugin which makes it easy to extract the 
+API documentation as part of a normal Sphinx workflow. This Sphinx plugin 
+is distributed with the source code for the CMake interpreter, and is 
+also available in a GitHub repository mirror [@sphinx_plugin]. 
+For completeness, we note that a couple of similar Sphinx
+plugins [@official_sphinx_domain; @marco_koch] also exist, but they appear
+to have been abandoned.
+
+To our knowledge, all of the aforementioned documentation solutions simply 
+extract the reST API documentation verbatim. Notably this means the
+developer is responsible for manually: adding the function/macro signatures 
+to the documentation, listing the function's parameters, and the overall
+formatting (*e.g.*, titles and subtitles, creation of parameter 
+tables, underline/overline consistency). For build system developers 
+maintaining one or two CMake modules, these are admittedly pretty minor 
+inconveniences; however, for an organizations which maintaining a substantial 
+CMake code base, these "minor inconveniences" can impact productivity. 
+In order to increase the productivity of the CMakePP team, we created CMinx.
+
+Concurrent with the submission of this manuscript, we have also released the 
+first production version of CMinx. Promisingly, the CMinx GitHub organization
```

Comment:
> first production version of CMinx

This would be a great place to put a reference to the PyPI page. We should probably also mention that it has a CMake interface somewhere and put a link to the GitHub page there